cms/djangoapps/contentstore/views/tests/test_transcript_settings.py

```diff
@@ -0,0 +1,167 @@
+        Allowed request methods include POST.
+        """
+        transcript_settings_url = self.get_url_for_course_key(self.course.id)
+        response = self.client.get(transcript_settings_url, content_type='application/json')
+        self.assertEqual(response.status_code, 405)
+
+    def test_404_with_feature_disabled(self):
+        """
+        Verify that 404 is returned if the corresponding feature is disabled.
+        """
+        transcript_settings_url = self.get_url_for_course_key(self.course.id)
+        with patch('openedx.core.djangoapps.video_config.models.VideoTranscriptEnabledFlag.feature_enabled') as feature:
+            feature.return_value = False
+            response = self.client.post(transcript_settings_url, content_type='application/json')
+            self.assertEqual(response.status_code, 404)
+
+    @ddt.data(
+        (
+            {
+                'provider': 'invalid_provider',
+                'api_key': '1234'
+            },
+            True,
+            400
+        ),
+        (
+            {
+                'provider': '3PlayMedia',
+                'api_key': '11111',
+                'api_secret_key': '44444'
+            },
+            False,
+            400
+        ),
+        (
+            {
+                'provider': 'Cielo24',
+                'api_key': '12345',
+                'username': 'test_user'
+            },
+            True,
+            200
+        )
+    )
+    @ddt.unpack
+    @patch('contentstore.views.transcript_settings.update_3rd_party_transcription_service_credentials')
+    def test_transcript_credentials_handler(self, request_payload, is_updated,
+                                            expected_status_code, mock_update_credentials):
+        """
+        Tests that transcript credentials handler works as expected.
+        """
+        mock_update_credentials.return_value = is_updated
+        transcript_settings_url = self.get_url_for_course_key(self.course.id)
+        response = self.client.post(
+            transcript_settings_url,
+            data=json.dumps(request_payload),
+            content_type='application/json'
+        )
+        self.assertEqual(response.status_code, expected_status_code)
+
+
+@ddt.ddt
+class TranscriptCredentialsValidationTest(TestCase):
+    """
+    Tests for credentials validations.
+    """
+
+    @ddt.data(
+        (
+            'ABC',
+            {
+                'username': 'test_user',
+                'password': 'test_pass'
+            },
+            'Invalid Provider ABC.',
+            {}
+        ),
+        (
+            'Cielo24',
+            {
+                'username': 'test_user'
+            },
+            'api_key must be specified.',
+            {}
+        ),
+        (
+            'Cielo24',
+            {
+                'username': 'test_user',
+                'api_key': 'test_api_key',
+                'extra_param': 'extra_value'
```

Comment:
Why are we adding extra param? if there is any need, we should add a comment

Reply:
I'll cover that in view docstring.